mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td

```diff
@@ -1084,8 +1084,79 @@ def LLVM_TBAATagAttr : LLVM_Attr<"TBAATag", "tbaa_tag"> {
   let assemblyFormat = "`<` struct(params) `>`";
 }
 
+def LLVM_TBAAStructFieldAttr : LLVM_Attr<"TBAAStructField", "tbaa_struct_field"> {
+  let parameters = (ins
+    "TBAANodeAttr":$typeDesc,
+    "int64_t":$offset,
+    "int64_t":$size
+  );
+  let assemblyFormat = "`<` struct(params) `>`";
+}
+
+
+def LLVM_TBAAStructFieldAttrArray : ArrayRefParameter<"TBAAStructFieldAttr"> {
```

Comment:
I wonder why we implemented our own array attributes at the time. @vzakhari do you remember? Potentially we could just use:
`OptionalArrayRefParameter<"TBAAStructFieldAttr">:$fields`
In LLVM_TBAATypeNodeAttr.

Reply:
Sorry, I do not remember.  It would probably work either way.

Reply:
@PikachuHyA  It is quite a lot of custom printing / parsing that could be saved with `OptionalArrayRefParameter<"TBAAStructFieldAttr">:$fields`. So maybe we want to give it a shot.